### ISSUE TITLE
still bad inversion but i dont know why

### ISSUE THREAD

**Issue by adeelnazar** [wty-ustc/e2style#22]
# still bad inversion but i dont know why

![celebrity-pics-3 (1) (1)](https://user-images.githubusercontent.com/29988547/208403662-0cef67b7-a8e2-4763-b9b8-7a783092578c.jpg)
![320937818_909805843710812_7225152858816556845_n (1) (1)](https://user-images.githubusercontent.com/29988547/208403675-f2bf5228-4d1d-4ccb-8bdb-ad97ab378f44.jpg)
![320925575_562852398514446_8652596532889929974_n (1) (1)](https://user-images.githubusercontent.com/29988547/208403678-180e2837-5968-42bf-8766-2ab68d630442.jpg)
![320505752_520230476748434_7768053947597212110_n (1) (1)](https://user-images.githubusercontent.com/29988547/208403683-aaac4d8f-c300-43ad-b723-2ef617c5fab0.jpg)
![320345500_931916501076882_7892912347158999581_n (2) (1)](https://user-images.githubusercontent.com/29988547/208403687-58d7c05d-bca1-4601-a39f-fbbb2f044f74.jpg)
![320345500_931916501076882_7892912347158999581_n (1) (1)](https://user-images.githubusercontent.com/29988547/208403693-13568b14-c64c-44a9-8275-794afb12a70e.jpg)
![4245574-tom-cruise (1) (1)](https://user-images.githubusercontent.com/29988547/208403696-b27b6781-5bcd-472b-b5a7-8fbd67a5fc5b.jpg)

**Comment by wtybest** [comment]
Use this [script](https://github.com/eladrich/pixel2style2pixel/blob/master/scripts/align_all_parallel.py) to align the images first.

**Comment by adeelnazar** [comment]
> Use this [script](https://github.com/eladrich/pixel2style2pixel/blob/master/scripts/align_all_parallel.py) to align the images first.

Pic of Angelina jolie and Tom cruise both pic are correct align but not good as result

**Comment by wtybest** [comment]
Can you send me an original image? I'll test it when I have a spare moment.

**Comment by adeelnazar** [comment]
> Can you send me an original image? I'll test it when I have a spare moment.

Give me Whatsapp no or email??

**Comment by wtybest** [comment]
You can comment the image directly here.

**Comment by adeelnazar** [comment]
> Y
![women-celebrity-megan-fox-looking-at-viewer-wallpaper-preview](https://user-images.githubusercontent.com/29988547/213653232-a672ee8c-0c09-45b8-a2cc-02f568aca90c.jpg)
![289433-man-face-human-portrait-look-men-person-ernst](https://user-images.githubusercontent.com/29988547/213653360-902f4a30-b9cc-4d0b-b3c2-1364d031822d.jpg)
![1640865-bigthumbnail](https://user-images.githubusercontent.com/29988547/213653459-4e13f77a-2794-4810-9d96-f563e7bac2b4.jpg)


ou can comment the image directly here.

**Comment by adeelnazar** [comment]
E2 Style I think it's best for inversion compare to others but for simple images but image with high make up and charming face are not good

**Comment by wtybest** [comment]
I agree. Our target inversion space is the W+ space of StyleGAN. Compared with peer work, our method is more superior. But for these high frequency features, inversion to the W+ space is not sufficient. See #15   for the reason and the solution.

**Comment by adeelnazar** [comment]
> I agree. Our target inversion space is the W+ space of StyleGAN. Compared with peer work, our method is more superior. But for these high frequency features, inversion to the W+ space is not sufficient. See #15 for the reason and the solution.

Is it possible to inversion image with HFGI inversion and style mixing with e2style with latentcode from HFGI
